### PR TITLE
Optimize container launch priority performance

### PR DESCRIPTION
**Pull request by FillZpp**
### Ⅰ. Describe what this PR does

Optimize container launch priority performance:

1. Make the controller reconcile the same pod and barrier less times.
2. Make the controller add the highest priority as soon as possible.
3. Add completed annotation to avoid pod to be reconciled again if a running pod turns to not ready.

### CODE REVIEW

File: pkg/controller/containerlaunchpriority/container_launch_priority_controller.go [hunk 1]
```diff
@@ -94,6 +93,30 @@ func add(mgr manager.Manager, r *ReconcileContainerLaunchPriority) error {
 	return nil
 }
 
+func shouldEnqueue(pod *v1.Pod, r client.Reader) bool {
```
Comment (furykerry) [inline pkg/controller/containerlaunchpriority/container_launch_priority_controller.go:96]: plz add ut for shouldEnqueue

File: apis/apps/pub/launch_priority.go [hunk 2]
```diff
@@ -29,4 +29,8 @@ const (
 	ContainerLaunchPriorityKey = "apps.kruise.io/container-launch-priority"
 	// ContainerLaunchOrdered is the annotation value that indicates containers in pod should be launched by ordinal.
 	ContainerLaunchOrdered = "Ordered"
+
+	// ContainerLaunchPriorityCompletedKey is the label indicates the pod has all its priorities
```
[the remark this answers is not in the document]
Comment (FillZpp) [reply apis/apps/pub/launch_priority.go]: fixed

File: pkg/util/containerlaunchpriority/container_launch_prirotiy.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package containerlaunchpriority
```
[the remark this answers is not in the document]
Comment (FillZpp) [reply pkg/util/containerlaunchpriority/container_launch_prirotiy.go:17]: fixed

File: pkg/controller/containerlaunchpriority/container_launch_priority_controller.go (hunk 1 above)
Comment (FillZpp) [reply pkg/controller/containerlaunchpriority/container_launch_priority_controller.go:96]: added

**Review by furykerry — COMMENTED** [review]
/lgtm

**Pull request merged**